vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc

```diff
@@ -0,0 +1,339 @@
+    return;
+  }
+
+  const std::string user_name = GetUserNameFromUrl(visit_data.path);
+  const std::string media_key = GetMediaKey(user_name);
+
+  if (media_key.empty()) {
+    OnMediaActivityError(visit_data, window_id);
+    return;
+  }
+
+  ledger_->GetMediaPublisherInfo(
+      media_key,
+      std::bind(&GitHub::OnMediaPublisherActivity,
+                this,
+                _1,
+                _2,
+                window_id,
+                visit_data,
+                media_key));
+}
+
+void GitHub::OnMediaPublisherActivity(
+    ledger::Result result,
+    ledger::PublisherInfoPtr info,
+    uint64_t window_id,
+    const ledger::VisitData& visit_data,
+    const std::string& media_key) {
+  if (result != ledger::Result::LEDGER_OK &&
+      result != ledger::Result::NOT_FOUND) {
+    OnMediaActivityError(visit_data, window_id);
+    return;
+  }
+  if (!info || result == ledger::Result::NOT_FOUND) {
+    const std::string user_name = GetUserNameFromUrl(visit_data.path);
+    const std::string url = GetProfileAPIURL(user_name);
+
+    FetchDataFromUrl(url,
+                     std::bind(&GitHub::OnUserPage,
+                               this,
+                               window_id,
+                               visit_data,
+                               _1,
+                               _2,
+                               _3));
+  } else {
+    GetPublisherPanelInfo(window_id,
+                          visit_data,
+                          info->id);
+  }
+}
+
+
+void GitHub::OnMediaActivityError(const ledger::VisitData& visit_data,
+                                        uint64_t window_id) {
+  std::string url = GITHUB_TLD;
+  std::string name = GITHUB_MEDIA_TYPE;
+
+  DCHECK(!url.empty());
+
+  ledger::VisitData new_visit_data;
+  new_visit_data.domain = url;
+  new_visit_data.url = "https://" + url;
+  new_visit_data.path = "/";
+  new_visit_data.name = name;
+
+  ledger_->GetPublisherActivityFromUrl(
+      window_id, ledger::VisitData::New(new_visit_data), std::string());
+}
+
+
+// Gets publisher panel info where we know that publisher info exists
+void GitHub::GetPublisherPanelInfo(
+    uint64_t window_id,
+    const ledger::VisitData& visit_data,
+    const std::string& publisher_key) {
+  auto filter = ledger_->CreateActivityFilter(
+    publisher_key,
+    ledger::EXCLUDE_FILTER::FILTER_ALL,
+    false,
+    ledger_->GetReconcileStamp(),
+    true,
+    false);
+  ledger_->GetPanelPublisherInfo(filter,
+    std::bind(&GitHub::OnPublisherPanelInfo,
+              this,
+              window_id,
+              visit_data,
+              publisher_key,
+              _1,
+              _2));
+}
+
+
+
+void GitHub::OnPublisherPanelInfo(
+    uint64_t window_id,
+    const ledger::VisitData& visit_data,
+    const std::string& publisher_key,
+    ledger::Result result,
+    ledger::PublisherInfoPtr info) {
+  if (!info || result == ledger::Result::NOT_FOUND) {
+    FetchDataFromUrl(visit_data.url,
```

Comment:
If I understand `OnUserPage` correctly, it's used as a response to github API. In this function we pass in `visit_data.url`. Is this API url as well? or just `https://github.com/username`

Reply:
Should be API but wasn't. Will fix